libs/wingc/src/type_check/lifts.rs

```diff
@@ -1,140 +1,93 @@
-use std::collections::{BTreeMap, BTreeSet};
+use std::collections::{BTreeMap, BTreeSet, HashMap};
 
-use itertools::Itertools;
-
-use crate::ast::Symbol;
+use crate::ast::{Symbol, UserDefinedType};
 
 use super::{ExprId, CLASS_INFLIGHT_INIT_NAME};
 
 /// A repository of lifts and captures at the class level.
 #[derive(Debug)]
 pub struct Lifts {
-	/// All the lifts. The key is "<method>/<token>"
-	lifts: BTreeMap<String, MethodLift>,
-
-	/// All the captures. The key is token.
-	captures: BTreeMap<String, Capture>,
+	// TODO: make all these private and add accessors+helper logic
+	/// All the lifts. Map from method to a map from inflight code to lift qualifications.
+	pub lifts_qualifications: BTreeMap<String, BTreeMap<String, LiftQualification>>,
 
-	/// Map from token to lift
-	lift_by_token: BTreeMap<String, Lift>,
+	/// All the captures. The key is token the value is the preflight code.
+	/// Used for preflight setup of inflight captures.
+	pub captures: BTreeMap<String, Capture>,
 
-	/// Map between expression id and a lift token.
-	token_by_expr_id: BTreeMap<ExprId, String>,
+	/// Map between liftable AST element and a lift token (used for inflight jsification of captures)
+	pub token_for_liftable: HashMap<Liftable, String>,
```

Comment:
Was the choice to use different map data structures intentional? (If so maybe add a comment, I'm not sure if the difference is obvious)

Reply:
In general I don't like our use of `BTreeMap` at all because it adds a ordering property that isn't needed and since we usually map from strings and ints we don't encounter an issue since ordering is implemented for them by default. In this case I didn't have or want to implement `Ord` on `Liftable`.
In cases where this affects output order I'm for using an `IndexedMap` which is ordered without the need for `Ord` either. But in this case the order has no implications on the output so I went for the least requirements solution (`HashMap`). I think there's no point in adding a comment here since moving back to a `BTreeMap` will give a complication error and ordering isn't important here.